src/data/polynomial/ring_division.lean

```diff
@@ -478,64 +486,68 @@ using_well_founded {dec_tac := tactic.assumption}
 
 /-- `roots p` noncomputably gives a multiset containing all the roots of `p`,
 including their multiplicities. -/
-noncomputable def roots (p : R[X]) : multiset R :=
+noncomputable def roots [decidable_eq R] (p : R[X]) : multiset R :=
 if h : p = 0 then ∅ else classical.some (exists_multiset_roots h)
```

Comment:
This change is unhelpful; the definition is already very non-computable, so there's no point adding this argument,
```suggestion
noncomputable def roots (p : R[X]) : multiset R :=
by letI := classical.dec_eq R; exact
  if h : p = 0 then ∅ else classical.some (exists_multiset_roots h)
```

Reply:
I've pushed a fix to that effect

Reply:
Thanks. I think that all the definitions on polynomials which require a `decidable_eq` instance should be obtained from `classical.dec_eq` as above, including for example `mv_polynomial.vars` in `data/mv_polynomial/variables.lean`. Otherwise, it escalates quickly... If you agree, I may restart this PR from scratch. (And eventually, the same could be done on the mathlib4 side.)

Reply:
I'm not sure I agree that they _should_ be, but at any rate they are already define that way. So the minimal fix is to add `classical.dec_eq` to all the relevant existing `def`s, and only add `decidable_eq` when need by theorem statements.